hive/src/main/java/org/apache/iceberg/hive/CustomHiveTableOperations.java

```diff
@@ -90,6 +90,11 @@ protected void doRefresh() {
           String errMsg = String.format("%s.%s is missing %s property", database, tableName, METADATA_LOCATION_PROP);
           throw new IllegalArgumentException(errMsg);
         }
+        if (!io().newInputFile(metadataLocation).exists()) {
```

Comment:
This is one more extra check we have to do. for every refresh. Once we are in stable state we should get rid of this.